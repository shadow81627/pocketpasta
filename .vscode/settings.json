{
  "eslint.validate": ["vue", "javascript", "javascriptreact"],
  "editor.formatOnSave": false,
  "vetur.validation.template": false,
  "files.exclude": {
    "**/node_modules/**": true,
    "**/dist/**": true,
    "**/coverage/**": true,
    "**/.nuxt/**": true,
    "**/sw.js": true,
    "**/.firebase/**": true
  },
  "editor.codeActionsOnSave": {
    "source.fixAll.eslint": true
  }
}
