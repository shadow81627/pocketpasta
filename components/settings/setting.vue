<template>
  <li class="list-group-item">
    <div class="row">
      <div class="col-sm-6">
        <v-card flat color="transparent">
          <v-card-title primary-title>
            <v-icon left>${{ icon }}</v-icon>
            <span :id="id" class="title font-weight-light">{{ label }}</span>
          </v-card-title>
          <v-card-text class="subheading form-text text-muted pt-0">{{
            description
          }}</v-card-text>
        </v-card>
      </div>
      <div class="col-sm-6 align-self-center">
        <slot :id="id">
          <input type="text" class="form-control" placeholder="placeholder" />
        </slot>
      </div>
    </div>
  </li>
</template>

<script>
export default {
  props: {
    icon: { type: String, default: null },
    label: { type: String, required: true },
    description: { type: String, default: null },
    id: { type: String, default: null },
  },
};
</script>
