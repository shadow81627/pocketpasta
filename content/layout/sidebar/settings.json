{
  "icon": "carbon:settings",
  "name": "Settings",
  "route": "settings",
  "pos": "1/3"
}
