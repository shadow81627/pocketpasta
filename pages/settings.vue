<template>
  <v-container>
    <v-row>
      <v-col>
        <h1>{{ $t('settings.heading') }}</h1>
      </v-col>
    </v-row>

    <!-- <v-expansion-panels multiple accordion>
      <v-expansion-panel>
        <v-expansion-panel-header>
          header
        </v-expansion-panel-header>
        <v-expansion-panel-content>
          content
        </v-expansion-panel-content>
      </v-expansion-panel>
      <v-expansion-panel>
        <v-expansion-panel-header>
          header
        </v-expansion-panel-header>
        <v-expansion-panel-content>
          content
        </v-expansion-panel-content>
      </v-expansion-panel>
    </v-expansion-panels> -->

    <v-row>
      <v-col>
        <v-card>
          <v-card-title>
            <h2>
              {{ $t('settings.style.heading') }}
            </h2>
          </v-card-title>

          <setting v-bind="$t('settings.style.theme-label')">
            <ThemeSelector aria-labelledby="theme-label" />
          </setting>
        </v-card>
      </v-col>
    </v-row>

    <v-row>
      <v-col>
        <v-card>
          <v-card-title>
            <h2>
              {{ $t('settings.content.heading') }}
            </h2>
          </v-card-title>
          <setting v-bind="$t('settings.content.language')">
            <LangSwitcher />
          </setting>
        </v-card>
      </v-col>
    </v-row>

    <v-row>
      <v-col>
        <v-card>
          <v-card-title>
            <h2>Database</h2>
          </v-card-title>
          <v-card-subtitle>Apache CouchDB Settings</v-card-subtitle>
          <v-container class="pt-0">
            <v-row>
              <v-col cols="12" class="align-self-center">
                <v-text-field v-model="dbRemote" label="Remote URL" />
              </v-col>
              <v-col cols="12" sm="6" class="align-self-center">
                <v-text-field v-model="dbUsername" label="Username" />
              </v-col>
              <v-col cols="12" sm="6" class="align-self-center">
                <v-text-field
                  v-model="dbPassword"
                  label="Password"
                  type="password"
                  autocomplete="one-time-code"
                />
              </v-col>
            </v-row>
          </v-container>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
import ThemeSelector from '@/components/ThemeSelector.vue';
import LangSwitcher from '@/components/Layout/LangSwitcher';
import setting from '@/components/settings/setting';

export default {
  components: {
    ThemeSelector,
    LangSwitcher,
    setting,
  },
  head() {
    return {
      link: [],
    };
  },
  computed: {
    dbRemote: {
      get() {
        return this.$warehouse.get('dbRemote') || this.$config.DB_REMOTE;
      },
      set(value) {
        this.$warehouse.set('dbRemote', value);
      },
    },
    dbUsername: {
      get() {
        return this.$warehouse.get('dbUsername') || this.$config.DB_USERNAME;
      },
      set(value) {
        this.$warehouse.set('dbUsername', value);
      },
    },
    dbPassword: {
      get() {
        return this.$warehouse.get('dbPassword') || this.$config.DB_PASSWORD;
      },
      set(value) {
        this.$warehouse.set('dbPassword', value);
      },
    },
  },
};
</script>
