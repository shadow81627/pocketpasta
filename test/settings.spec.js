import settings from '@/pages/settings.vue';
import { shallowMount } from '@vue/test-utils';

const factory = () => shallowMount(settings, {});

describe('settings', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('head', () => {
    expect(settings.head()).toBeTruthy();
  });
});
