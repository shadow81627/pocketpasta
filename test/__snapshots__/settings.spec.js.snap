// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings renders properly 1`] = `
"<div class=\\"container\\"><div class=\\"row\\"><div class=\\"col\\"><h1>settings.heading</h1></div></div> <div class=\\"card mb-3\\"><h2 class=\\"subheading card-header text-uppercase\\">
      settings.style.heading
    </h2> <ul class=\\"list-group list-group-flush\\"><setting-stub><theme-selector-stub aria-labelledby=\\"theme-label\\"></theme-selector-stub></setting-stub></ul></div> <div class=\\"card mb-3\\"><h2 class=\\"subheading card-header text-uppercase\\">
      settings.content.heading
    </h2> <ul class=\\"list-group list-group-flush\\"><setting-stub><lang-switcher-stub></lang-switcher-stub></setting-stub></ul></div></div>"
`;
